### PR TITLE
Add TRex 1.5.0

### PR DESCRIPTION
**Pull request by melonamin**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/trex.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  url "https://github.com/amebalabs/TRex/releases/download/v#{version}/TRex.zip",
+      verified: "github.com/amebalabs/TRex/releases/"
+  name "TRex"
+  desc "Easy to use text extraction tool"
+  homepage "https://trex.ameba.co/"
+
+  livecheck do
+    url :url
+    strategy :github_latest
+  end
```
Comment (neutric) [inline Casks/trex.rb:17]: Did you have a specific reason to use `strategy :github_latest`? `strategy :github_latest` comes at the cost of a limited API access and is only necessary if there likely will be pre-releases or tags that don't include binaries. If not, the whole `livecheck` stanza can go in favor of the built-in, implicit `livecheck` for GitHub repos.

**Review by neutric — COMMENTED** [review]
Thanks for your contribution to `homebrew-cask`, @melonamin! My review is in the code. Take it with a grain of salt as I am not a maintainer.

File: Casks/trex.rb (hunk 1 above)
Comment (melonamin) [reply Casks/trex.rb:17]: No, I'm actually not sure about this. I've copied this from another formula I maintain and I believe that one of the maintainers asked to add it.  

Let me remove it.

File: Casks/trex.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# typed: false
+# frozen_string_literal: true
+
```
[the remark this answers is not in the document]
Comment (melonamin) [reply Casks/trex.rb:3]: OK.

But this was added automatically by brew style

File: Casks/trex.rb (hunk 1 above)
Comment (melonamin) [reply Casks/trex.rb:17]: > will be pre-releases or tags that don't include binaries. 

On a second read, I do have  which I don't want to expose to Homebrew. Seems like `strategy :github_latest` is a better fit

File: Casks/trex.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (neutric) [reply Casks/trex.rb]: You might have hit a corner case in your workflow, then. I believe this behavior is not intended: https://github.com/Homebrew/brew/blob/master/Library/.rubocop.yml#L468-L477. These two comments are missing from all but one cask definition, and there are close to 4000 now.

File: Casks/trex.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+cask "trex" do
+  version "1.5.0"
+  sha256 "035ffda6acc996002518f9731bb566fe3de6fb840ba0cd9a7b6722f3a9f608c2"
+
+  url "https://github.com/amebalabs/TRex/releases/download/v#{version}/TRex.zip",
+      verified: "github.com/amebalabs/TRex/releases/"
```
[the remark this answers is not in the document]
Comment (neutric) [reply Casks/trex.rb:6]: I wasn't clear enough: the trailing `releases/`should be deleted as `verified` should contain "the smallest possible portion of the URL that uniquely identifies the app or vendor" according to the [Cask Cookbook](https://docs.brew.sh/Cask-Cookbook#when-url-and-homepage-domains-differ-add-verified).

**Comment by miccal** [comment]
Thank you @melonamin.

**Comment by alexwhitman** [comment]
Running `brew upgrade` is erroring on TRex:
```text
$ brew upgrade
==> Casks with 'auto_updates true' or 'version :latest' will not be upgraded; pass `--greedy` to upgrade them.
==> Upgrading 1 outdated package:
melonamin/formulae/trex v1.2 -> v1.4.3
==> Upgrading trex
==> Downloading https://github.com/amebalabs/TRex/releases/download/v1.4.3/TRex.zip
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/340433391/2560d5d0-911e-4e48-bb86-cff3811248b4?X-Amz-Algorith
######################################################################## 100.0%
==> Purging files for version v1.4.3 of Cask trex
Error: melonamin/formulae/trex: SHA256 mismatch
Expected: e18a030af4b1037022736c12045a6eff8aa13b65201c9ef98d0a7f36d050dc36
  Actual: f231a866ba1fe2e85e33bf81c33b5fc0f63548857dca54cbd7d8d60d1ef4cbdf
    File: /Users/alex/Library/Caches/Homebrew/downloads/1b211f92cb40684d26b3ac2d2e8dbf7901552505aba207f85b01a8f1233a0cff--TRex.zip
To retry an incomplete download, remove the file above.
```
This is after a `brew update`. For some reason it's downloading 1.4.3 rather than 1.5.0 and erroring on the SHA356 hash. Can anyone with more `brew` knowledge than myself advise?

**Pull request merged**